cmd/stacker/publish.go

```diff
@@ -104,17 +108,18 @@ func doPublish(ctx *cli.Context) error {
 	}
 
 	args := stacker.PublishArgs{
-		Config:     config,
-		ShowOnly:   ctx.Bool("show-only"),
-		Substitute: ctx.StringSlice("substitute"),
-		Tags:       ctx.StringSlice("tag"),
-		Url:        ctx.String("url"),
-		Username:   ctx.String("username"),
-		Password:   ctx.String("password"),
-		Force:      ctx.Bool("force"),
-		Progress:   shouldShowProgress(ctx),
-		SkipTLS:    ctx.Bool("skip-tls"),
-		LayerTypes: layerTypes,
+		Config:         config,
+		ShowOnly:       ctx.Bool("show-only"),
+		Substitute:     ctx.StringSlice("substitute"),
+		SubstituteFile: ctx.String("substitute-file"),
```

Comment:
While --substitute-file is an awesome feature, it probably should have been a separate patch (so one feature can be reverted without the other if needed)

Reply:
substitute-file support already in stacker but for build cmd, now added to publish cmd. 
convert cmd and test cases less useful without that.
But split into separate commits now.